### PR TITLE
cairo: add cairo-xlib component

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **cairo/***

this component is required by gtk

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by Croydon — DISMISSED** [review]
https://github.com/freedesktop/cairo/blob/ab7b45feb701032c73ef0c190d963ee3ff76a815/meson.build#L250

File: recipes/cairo/all/test_package/conanfile.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 class TestPackageConan(ConanFile):
     settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake"
+    generators = "cmake", "pkg_config"
```
Comment (SpaceIm) [inline recipes/cairo/all/test_package/conanfile.py:7]:
```suggestion
    generators = "cmake", "pkg_config"
```

pkg_config is useless here

**Review by Croydon — APPROVED** [review]

**Comment by SpaceIm** [comment]
What about:
- cairo-png and cairo-svg: https://github.com/freedesktop/cairo/blob/ab7b45feb701032c73ef0c190d963ee3ff76a815/meson.build#L159
- cairo-win32-font: https://github.com/freedesktop/cairo/blob/ab7b45feb701032c73ef0c190d963ee3ff76a815/meson.build#L406
- cairo-script: https://github.com/freedesktop/cairo/blob/ab7b45feb701032c73ef0c190d963ee3ff76a815/meson.build#L619
- cairo-ps: https://github.com/freedesktop/cairo/blob/ab7b45feb701032c73ef0c190d963ee3ff76a815/meson.build#L628
- cairo-xml: https://github.com/freedesktop/cairo/blob/ab7b45feb701032c73ef0c190d963ee3ff76a815/meson.build#L650

**Comment by ericLemanissier** [comment]
feel free to make a PR (on my branch or on CCI directly), that's orthogonal

**Review by SSE4 — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Comment by Vhab** [comment]
@ericLemanissier This change breaks building with 'with_xlib=False',

`ConanException: cairo/1.17.2 package_info(): Package require 'xorg' declared in components requires but not defined as a recipe requirement`

**Comment by prince-chrismc** [comment]
fix in #3666

**Pull request merged**